src/components/screens/editor/editor-screen.tsx

```diff
@@ -182,6 +193,283 @@ const EditorScreen: React.FC<Props> = ({
+      intentId: link.getID()
+    });
+
+    setStoryData({
+      ...storyData,
+      selectedIntent: intent,
+      selectedKnot: undefined
+    });
+  }
+
+  /**
+   * Event handler for set active training material change 
+   * 
+   * @param event event
+   */
+  const onSetActiveTrainingMaterialChange = (event: any) => {
+    const { name, value } = event.target;
+    if (!name || !value || !trainingMaterial || !selectedIntent || !intents) {
+      return;
+    }
+    let foundMaterial: TrainingMaterial | undefined;
+    if (value === "none") {
+      setEditedTrainingMaterial(undefined);
+    } else {
+      foundMaterial = trainingMaterial.find(item => item.id === value);
+      foundMaterial && setEditedTrainingMaterial(foundMaterial);
+    }
+    const key = objectKeyConversion(name);
+    selectedIntent.trainingMaterials = { ...selectedIntent.trainingMaterials, [key]: foundMaterial?.id ?? undefined };
+
+    if (accessToken && selectedIntent.id) {
+      Api.getIntentsApi(accessToken).updateIntent({
+        intentId: selectedIntent.id,
+        intent: selectedIntent,
+        storyId: storyId
+      });
+    }
+    
+    
+    setStoryData({
+      ...storyData,
+      selectedIntent: selectedIntent,
+      intents: [ ...intents.filter(item => item.id !== selectedIntent.id), selectedIntent ]
+    });
+  }
+
+  /**
+   * Event handler for add training material click
+   * 
+   * @param name name of the training material type
+   */
+  const onAddTrainingMaterialClick = (name: any) => {
+    if (!selectedIntent?.id) {
+      return;
+    }
+
+    setEditingTrainingMaterial(true);
+    setSelectedTrainingMaterialType(name as keyof object);
+    setEditedTrainingMaterial({
+      type: name,
+      storyId: storyId,
+      name: "",
+      text: "",
+      visibility: TrainingMaterialVisibility.STORY
+    });
+  }
+
+  /**
+   * Event handler for edit training material click
+   */
+  const onEditTrainingMaterialClick = () => {
+    if (!editedTrainingMaterial?.type) {
+      return;
+    }
+    setEditingTrainingMaterial(true);
+    setSelectedTrainingMaterialType(editedTrainingMaterial.type);
+  }
+
+  /**
+   * Event handler for delete training material click
+   */
+  const onDeleteTrainingMaterialClick = () => {
+    if (!editedTrainingMaterial?.type || !selectedIntent || !intents) {
+      return;
+    }
+
+    const key = objectKeyConversion(editedTrainingMaterial.type);
+
+    setStoryData({
+      ...storyData,
+      selectedIntent: { ...selectedIntent, trainingMaterials: { ...selectedIntent.trainingMaterials, [key]: undefined } },
+      intents: intents.map(item => item.id === selectedIntent.id ? selectedIntent : item)
+    });
+    setEditingTrainingMaterial(false);
+    setEditedTrainingMaterial(undefined);
+  }
+
+  /**
+   * Event handler for save training material click
+   * 
+   * @param action action type: update / create
+   */
+  const onSaveTrainingMaterialClick = async (action: string) => {
```

Comment:
This can and should be separated to create and update methods, without any parameters. The correct method to call can be inferred in where they should be called.

Reply:
Removed parameter, now executing action decision in editor-screen.